UniversalAlgebra/Structures/Iso.lagda

```diff
@@ -0,0 +1,298 @@
+---
+layout: default
+title : Structures.Iso module (The Agda Universal Algebra Library)
+date : 2021-06-22
+author: [the ualib/agda-algebras development team][]
+---
+
+### <a id="isomorphisms">Isomorphisms</a>
+
+\begin{code}
+
+{-# OPTIONS --without-K --exact-split --safe #-}
+
+module Structures.Iso where
+
+
+-- Imports from Agda (builtin/primitive) and the Agda Standard Library ---------------------
+open import Agda.Primitive                        using    ( _⊔_    ;   lsuc     )
+                                                  renaming ( Set    to  Type     )
+open import Agda.Builtin.Equality                 using    ( _≡_    ;   refl     )
+open import Data.Product                          using    ( _,_    ;   Σ-syntax
+                                                           ;  Σ     ;   _×_      )
+                                                  renaming ( proj₁  to  fst
+                                                           ; proj₂  to  snd      )
+open import Level                                 using    ( Level  ;  Lift
+                                                           ; lift   ;  lower     )
+open import Function.Base                         using    ( _∘_                 )
+open import Relation.Binary.PropositionalEquality using    ( cong   ; cong-app   )
+
+
+-- Imports from agda-algebras --------------------------------------------------------------
+open import Overture.Preliminaries using ( ∣_∣ ; _≈_ ; ∥_∥ ; _∙_ ; lower∼lift ; lift∼lower )
+open import Structures.Basic         using ( Signature ; Structure ; Lift-Struc )
+open import Structures.Homs         using ( Hom ; 𝒾𝒹 ; ∘-Hom ; 𝓁𝒾𝒻𝓉 ; 𝓁ℴ𝓌ℯ𝓇 )
+
+private variable 𝑅 𝐹 : Signature
+
+\end{code}
+
+#### <a id="isomorphism-toolbox">Definition of isomorphism</a>
+
+Recall, `f ~ g` means f and g are *extensionally* (or pointwise) equal; i.e., `∀ x, f x ≡ g x`. We use this notion of equality of functions in the following definition of **isomorphism**.
+
+\begin{code}
+
+module _ {α ρᵃ β ρᵇ : Level} where
+
+ _≅_ : (𝑨 : Structure {α}{ρᵃ} 𝑅 𝐹)(𝑩 : Structure {β}{ρᵇ} 𝑅 𝐹) → Type (α ⊔ ρᵃ ⊔ β ⊔ ρᵇ)
+ 𝑨 ≅ 𝑩 =  Σ[ f ∈ (Hom 𝑨 𝑩)] Σ[ g ∈ Hom 𝑩 𝑨 ] ((∣ f ∣ ∘ ∣ g ∣ ≈ ∣ 𝒾𝒹 𝑩 ∣) × (∣ g ∣ ∘ ∣ f ∣ ≈ ∣ 𝒾𝒹 𝑨 ∣))
```

Comment:
This 4-field thing really screams "record". I have written a lot of code with such equivalences, and don't regret using records at all.

Reply:
Sounds good. I'll try that.